### PR TITLE
feat(Stryker CLI 'init'): Support for preset configuration during 'stryker init'

### PR DESCRIPTION
**Pull request by Wmaarts**
## Implementation of #1049 

```
? Are you using one of these frameworks? Then select a preset configuration. (Use arrow keys)
> angular
  react
  vueJs
  ──────────────
  none
```

#### Small disclaimer
(my first time contributing to an open-source project, hope I'm doing this right. Also, I'm new to the stryker project. 😄 )

## Summary
- Running 'stryker init' prompts the user to select a preset (or do custom setup).
- Supports Angular, React and vuejs
- Loads a fixed configuration and installs appropriate dependencies
- Supports adding preset-specific questions
  - For example: React preset prompts the user to select between JSX and TSX config
- Some unit tests written
- All based on the [the stryker handbook](https://github.com/stryker-mutator/stryker-handbook)

### Where?
Presets are defined in the stryker package under \src\initializer\presets.
### Adding a new preset
Extends the StrykerPreset class like so:
```
export class MyPreset extends StrykerPreset { 
    public dependencies: string[] = [];        // preset-specific dependencies here
    public conf: string = '';                  // will be loaded into stryker.conf.js
    public async prompt(): Promise<void> { };  // will be called after selecting this preset
}
```
Add your preset to StrykerPresets.ts like so:
```
const strykerPresets: PresetOption[] = [
        {name: 'angular', create() { return new AngularPreset(); } },
        {name: 'react', create() { return new ReactPreset(); } },
        {name: 'vueJs', create() { return new VueJsPreset(); } },
        {name: 'myPreset', create() { return new MyPreset(); } } // <- add this line
        // Add new presets here
];
``` 
Done. 👍 

### Shortcomings
- Has only been fully tested on a small Angular project (I had nothing else at hand... )
- No integration tests yet
- (nitpick) When using angular, the tests should be run using `npx stryker run` instead of `stryker run`, but when the angular preset is chosen the user is still told to use `stryker run`.

### CODE REVIEW

File: packages/stryker-javascript-mutator/package.json [hunk 1]
```diff
@@ -47,7 +47,6 @@
   "devDependencies": {
     "@babel/types": "~7.1.2",
     "@types/babel__generator": "^7.0.0",
-    "@types/babel__parser": "^7.0.0",
```
[the remark this answers is not in the document]
Comment (Wmaarts) [reply packages/stryker-javascript-mutator/package.json]: Had to do with this PR (https://github.com/stryker-mutator/stryker/pull/1243) and also (https://github.com/stryker-mutator/stryker/pull/1247). These changes were needed in order to make 'npm install' work and they accidentally ended up in this PR too.
I merged stryker/master into my fork branch, I think that fixes it.

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "@types/mkdirp": "^0.5.2",
     "@types/mocha": "^2.2.44",
     "@types/rimraf": "2.0.2",
-    "@types/sinon": "^5.0.1",
+    "@types/sinon": "5.0.5",
```
[the remark this answers is not in the document]
Comment (Wmaarts) [reply package.json]: same as above comment

File: packages/stryker/package.json [hunk 3]
```diff
@@ -83,7 +83,10 @@
     "@types/node": "~10.11.4",
     "@types/prettier": "~1.13.1",
     "@types/progress": "~2.0.1",
-    "stryker-api": "^0.21.5"
+    "stryker-api": "^0.21.5",
+    "stryker-html-reporter": "^0.16.7",
```
[the remark this answers is not in the document]
Comment (Wmaarts) [reply packages/stryker/package.json]: I'll remove then, yes. These were accidentally generated while testing stryker init.

File: packages/stryker/src/initializer/StrykerPresets.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { AngularPreset } from './presets/AngularPreset';
+import { VueJsPreset } from './presets/VueJsPreset';
+import { ReactPreset } from './presets/ReactPreset';
+import PresetOption from './PresetOption';
+
+const strykerPresets: PresetOption[] = [
+        {name: 'angular', create() { return new AngularPreset(); } },
```
[the remark this answers is not in the document]
Comment (Wmaarts) [reply packages/stryker/src/initializer/StrykerPresets.ts]: The idea was to preserve memory by only instantiating a preset once it's needed. Hence the arrow functions.

[the remark this answers is not in the document]
Comment (nicojs) [reply packages/stryker/src/initializer/StrykerPresets.ts]: I understand, but you'll save a couple of bytes at most. I would rather have a piece of code that is better to understand and easier to maintain. 

Ever heard of the phrase:

> [Premature optimization is the root of all evil](http://wiki.c2.com/?PrematureOptimization)

[the remark this answers is not in the document]
Comment (Wmaarts) [reply packages/stryker/src/initializer/StrykerPresets.ts]: > but you'll save a couple of bytes at most.
Fair point. 😅 
I implemented your suggestion [here](https://github.com/stryker-mutator/stryker/pull/1248/commits/bdbb16a5f7559e85654e9742c20eb8d5566fd950#diff-49f55db075f2c1c2d325ab6bdf1f5c23R7).

**Comment by Wmaarts** [comment]
Pushed some changes that were suggested, but I'm not sure why the build is failing now.

**Comment by nicojs** [comment]
@Wmaarts 
>  I'm not sure why the build is failing now.

I don't know either. I've merged master and pushed to your branch. New build is running.

**Comment by nicojs** [comment]
I've committed a few more changes. Namely renamed some stuff, changed the abstract classes to interfaces where appropriate and added the handbook url to the outputted stryker.conf.js file.

**Comment by nicojs** [comment]
Thanks again for your help @Wmaarts

**Pull request merged**